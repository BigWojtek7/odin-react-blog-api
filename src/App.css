body {
  margin: 3vw 10vw;
  background-color: #d2d2d2;
}

#root {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
  font-family: sans-serif, 'Times New Roman';
}

.content {
  background-color: #bcbcbc;
  border-radius: 20px;
  padding: 0 5vw 5vw 5vw;
}

.card {
  padding: 2em;
}

.read-the-docs {
  color: #888;
}

form {
  display: flex;
  flex-direction: column;
  align-items: flex-start;
}

input,
textarea {
  margin-bottom: 10px;
}

textarea {
  width: 300px;
  height: 150px;
  resize: none;
}

button{
  cursor: pointer;
  border: none;
  border-radius: 10px;
  color: rgb(0, 126, 217);
  padding: 5px 10px;
  font-size: 1rem;
  font-weight: 700;
}