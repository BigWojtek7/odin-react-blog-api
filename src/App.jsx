import MainLayout from './layouts/MainLayout';
import AuthProvider from './contexts/AuthProvider';
import LoaderProvider from './contexts/LoaderProvider';
import ModalProvider from './contexts/ModalProvider';
import NotificationProvider from './contexts/NotificationProvider';

function App() {
  return (
    <>
      <NotificationProvider>
        <ModalProvider>
          <LoaderProvider>
            <AuthProvider>
              <MainLayout />
            </AuthProvider>
          </LoaderProvider>
        </ModalProvider>
      </NotificationProvider>
    </>
  );
}

export default App;
