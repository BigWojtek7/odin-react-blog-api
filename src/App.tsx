import MainLayout from './layouts/MainLayout';
import AuthProvider from './contexts/Auth/AuthProvider';
import LoaderProvider from './contexts/Loader/LoaderProvider';
import ModalProvider from './contexts/Modal/ModalProvider';
import NotificationProvider from './contexts/Notification/NotificationProvider';

function App() {
  return (
    <>
      <NotificationProvider>
        <ModalProvider>
          <LoaderProvider>
            <AuthProvider>
              <MainLayout />
            </AuthProvider>
          </LoaderProvider>
        </ModalProvider>
      </NotificationProvider>
    </>
  );
}

export default App;
