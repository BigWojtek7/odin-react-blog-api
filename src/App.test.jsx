describe('something truthy and falsy', () => {
  it('true to be true', () => {
    expect(true).toBe(true);
  });

  it('false to be false', () => {
    expect(false).toBe(false);
  });
});

import { render, screen } from '@testing-library/react';

import App from './App';
import { BrowserRouter } from 'react-router-dom';

describe('App', () => {
  it('renders whole app', () => {
    const { container } = render(
      <BrowserRouter>
        <App />
      </BrowserRouter>
    );
    expect(container).toMatchSnapshot();
  });
});
