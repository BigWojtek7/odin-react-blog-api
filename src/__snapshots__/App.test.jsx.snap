// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App > renders whole app 1`] = `
<div>
  <div
    class="content"
  >
    <header>
      <div
        class="_userBar_63c2ea _container_01e181"
      >
        <div
          class="_accountLinks_63c2ea"
        >
          <a
            href="/login"
          >
            Log-in
            <svg
              role="presentation"
              style="width: 1.5rem; height: 1.5rem;"
              viewBox="0 0 24 24"
            >
              <path
                d="M11 7L9.6 8.4L12.2 11H2V13H12.2L9.6 15.6L11 17L16 12L11 7M20 19H12V21H20C21.1 21 22 20.1 22 19V5C22 3.9 21.1 3 20 3H12V5H20V19Z"
                style="fill: currentColor;"
              />
            </svg>
          </a>
          <a
            href="/sign-up"
          >
            Sign Up 
            <svg
              role="presentation"
              style="width: 1.5rem; height: 1.5rem;"
              viewBox="0 0 24 24"
            >
              <path
                d="M15,14C12.33,14 7,15.33 7,18V20H23V18C23,15.33 17.67,14 15,14M6,10V7H4V10H1V12H4V15H6V12H9V10M15,12A4,4 0 0,0 19,8A4,4 0 0,0 15,4A4,4 0 0,0 11,8A4,4 0 0,0 15,12Z"
                style="fill: currentColor;"
              />
            </svg>
          </a>
        </div>
      </div>
      <hr
        class="_hr_5c36da"
      />
      <div
        class="_logoContainer_5c36da"
      >
        <a
          class="_logo_5c36da"
          href="/"
        >
          Daily Blog
        </a>
      </div>
      <hr
        class="_hr_5c36da"
      />
    </header>
    <main
      class="_main_ccfe8f"
    />
  </div>
</div>
`;
